contrib/k8s/README.md

```diff
@@ -1,41 +1,123 @@
+The other assumption is that your Kubernetes cluster will be routable on `172.17.4.99`  (which is what it will be if you use [coreos-kubernetes single node][coreos-kubernetes-single-node], and the issuer URL for your dex installation is `http://172.17.4.99:30556`; in production installations you will need to make sure that you are serving on https and you will likely want a host name on the normal TLS port instead of an IP address.
 
-```
-ssh worker
-cat > /proc/$(pgrep kubelet)/cwd/.dockercfg
-```
+[coreos-kubernetes-single-node][https://github.com/coreos/coreos-kubernetes/blob/master/single-node/README.md]
+
+## Start Postgres
 
-Start postgres
+Dex needs a database to store information; these commands will create a Postgres service that dex can use. Note that this configuration is not suitable for production - if the container is destroyed, the data is gone forever. 
+
+In production you should have a sufficiently fault-tolerant Postgres deployment on a persistent volume with backup.
 
 ```
 kubectl create -f postgres-rc.yaml
 kubectl create -f postgres-service.yaml
 ```
 
-Run dex and setup services
+## Create your secrets.
+
+dex needs a secret key for encrypting private keys in the database. These can be stored as [Kubernetes secrets][k8s-secrets].
+
+[k8s-secrets]: http://kubernetes.io/v1.0/docs/user-guide/secrets.html
+
+```
+kubectl create -f dex-secrets.yaml
+```
+
+## Start the Overlord
+
+Start the overlord. This will also initialize your database the first time it's run, and perform migrations when new versions are installed.
+
+```
+kubectl create -f dex-overlord-rc.yaml
+kubectl create -f dex-overlord-service.yaml
+```
+
+Note: this will expose the admin API on the cluster, which is essentially all-powerful. This will be fixed very soon. 
+
+## Add a Connector
+
+This is bit of a hack; right now the only way to add connectors and register
+your first client is to use the `dexctl` tool talking directly to the
+database. Because we the database is only routable inside the cluster, we do it
+inside a pod via `kubectl exec`.
+
+The other hacky thing is that this needs to happen before the workers start because workers do not (yet!) respond dynamically to connector configuration changes.
+
+First, start a shell session on the overlord pod.
+```
+DEX_OVERLORD_POD=$(kubectl get pod -l=app=dex,role=overlord -o template -t "{{ (index .items 0).metadata.name }}")
+
+kubectl exec -ti $DEX_OVERLORD_POD -- sh
+```
+
+Once we're on the pod, we create a connectors file and upload it to dex.
 
 ```
-for i in dex-overlord-rc.yaml dex-overlord-service.yaml dex-worker-rc.yaml dex-worker-service.yaml; do 
-	kubectl create -f ${i}
-done
+DEX_CONNECTORS_FILE=$(mktemp  /tmp/dex-conn.XXXXXX)
+cat << EOF > $DEX_CONNECTORS_FILE
+[
+	{
+		"type": "local",
+		"id": "local"
+	}
+]
+EOF
+
+/opt/dex/bin/dexctl --db-url=postgres://postgres@dex-postgres.default:5432/postgres?sslmode=disable set-connector-configs $DEX_CONNECTORS_FILE
+exit
```

Comment:
Nit: I think this could be one-lined rather than heredoc to tmpfile. But if it works, it works.